src/code/z_camera.c

```diff
@@ -1,4 +1,95 @@
+ *
+ * You should also be familiar with Euler angles: 'pitch', 'yaw', and 'roll'.
+ *      pitch: rotation about the X-axis, measured from +Y.
+ *          Unlike yaw and roll, pitch is bounded in +-0x4000 (90 degrees).
+ *          Pitch is 0 when the camera points parallel to the xz-plane (+Y points straight up).
+ *
+ *      yaw: rotation about the Y-axis, measured from (absolute) +Z.
+ *          Positive yaw rotates clockwise, towards +X.
+ *
+ *      roll: rotation about the Z-axis, measured from the camera's right direction.
+ *          Unfortunately, it's weird: For some reason, roll is flipped. Positive roll makes the camera
+ *          rotate counterclockwise, which means the WORLD rotates clockwise. Luckily roll is rarely
+ *          used.
+ *
+ *      Remember the right hand rule: make a thumbs-up with your right hand, stick your thumb in the
+ *      +direction (except for roll), and the angle follows the rotation of your curled fingers.
+ *
+ * Illustrations:
+ * Following the right hand rule, each hidden axis's positive direction points out of the screen.
+ *
+ *       YZ-Plane (pitch)        XZ-Plane (yaw)          XY-Plane (roll -- Note flipped)
+ *          +Y                      -Z                      +Y
+ *           ^                       ^ (into the             ^
+ *         --|--                     |   screen)             |<-
+ * +pitch /  |  \ -pitch             |                       |  \ -roll
+ *       v   |   v                   |                       |   |
+ * +Z <------O------> -Z   -X <------O------> +X   -X <------O------> +X
+ *           |                   ^   |   ^                   |   |
+ *           |                    \  |  /                    |  / +roll
+ *           |               -yaw  --|--  +yaw               |<-
+ *           v                       v                       v
+ *          -Y                      +Z                      -Y
+ *
+ */
+
+void func_800DDFE0(Camera* camera);
+s32 Camera_ChangeMode(Camera* camera, s16 mode);
+s16 Camera_ChangeSettingFlags(Camera* camera, s16 setting, s16 flags);
+s16 Camera_UnsetStateFlag(Camera* camera, s16 flags);
+
+#include "z_camera_data.c"
+
+PlayState* sCamPlayState;
+SwingAnimation D_801EDC30[4];
+Vec3f D_801EDDD0;
+Vec3f D_801EDDE0;
+Vec3f D_801EDDF0;
+
+// Camera will reload its paramData. Usually that means setting the read-only data from what is stored in
+// CameraModeValue arrays. Although sometimes some read-write data is reset as well
+#define RELOAD_PARAMS(camera) (camera->animState == 0 || camera->animState == 10 || camera->animState == 20)
+
+/**
+ * Camera data is stored in both read-only data and OREG as s16, and then converted to the appropriate type during
+ * runtime. If a small f32 is being stored as an s16, it is common to store that value 100 times larger than the
+ * original value. This is then scaled back down during runtime with the CAM_DATA_UNSCALE macro.
+ */
+#define CAM_DATA_SCALE(x) ((x)*100.0f)
```

Comment:
Would it make sense to name this in a way that signifies that it's a conversion for storage to a different type?

Reply:
Yeah, your original comment makes sense. I'm still trying to work out all the types and usages of this. There's just this factor of 100 and 0.01 that they throw everywhere in camera, and I'm still trying to nail down the advantage of that. Types seems like a good reason but I still haven't 100% investigated

Reply:
I ended up going with `CAM_RODATA_SCALE` since it does seem to be used for the rodata mostly